src/Core/Queues/IQueue.cs

```diff
@@ -21,10 +21,16 @@ public interface IQueue<T> : IHaveSerializer, IDisposable where T : class {
 
         Task<QueueEntry<T>> DequeueAsync(CancellationToken cancellationToken = default(CancellationToken));
 
+        [Obsolete("Use QueueEntry<T> overload")]

```

Comment:
Thanks for your pull request!!  Is there any chance you can revert these changes to the interface. I'm not remembering specifics, but we tried to go this route and ended up reverting to just take a string id.